### PR TITLE
make sure that we update the unencrypted size for the versions

### LINKED ISSUE #20234: File versioning seems broken
### Steps to reproduce
1. Create a local file (tested with docx and pdf, txt files seems to work fine)
2. Let the client sync, write something, save...produce some versions on the server
3. Login into the webinterface, click on the file -> versions
4. Try to download or restore it

### Expected behaviour
Versions should be downloadable and restorable.

### Actual behaviour
Download won't start, the restore brings a broken file to the client or a "Verbindung beendet" / "Broken webserver returning empty content length for non-empty file on resume".

### Server configuration
**Operating system**:
Ubuntu 14.04.3 LTS

**Web server:**
Apache  2.4.7
**Database:**
MySQL 5.5.46
**PHP version:**
PHP5 5.5.9
**ownCloud version:** (see ownCloud admin page)
8.2.0
**Updated from an older ownCloud or fresh install:**
updated
**List of activated apps:**
Enabled:
  - calendar: 0.8.0
  - contacts: 0.5.0.0
  - encryption: 1.1.0
  - files: 1.2.0
  - files_locking:
  - files_pdfviewer: 0.7
  - files_sharing: 0.7.0
  - files_texteditor: 2.0
  - files_trashbin: 0.7.0
  - files_versions: 1.1.0
  - files_videoviewer: 0.1.3
  - gallery: 14.2.0
  - mozilla_sync: 1.4

**The content of config/config.php:**
<?php
$CONFIG = array (
  'logtimezone' => 'Europe/Berlin',
  'instanceid' => 'XXX',
  'passwordsalt' => 'XXX',
  'datadirectory' => ''/var/www/owncloud/data'',
  'dbtype' => 'mysql',
  'version' => '8.2.0.12',
  'dbname' => 'XXX',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'XXX',
  'dbpassword' => 'XXX',
  'installed' => true,
  'forcessl' => true,
  'maxZipInputSize' => 0,
  'allowZipDownload' => true,
  'loglevel' => 2,
  'theme' => '',
  'maintenance' => false,
  'mail_smtpmode' => 'smtp',
  'mail_smtphost' => '127.0.0.1',
  'mail_smtpport' => '25',
  'mail_smtptimeout' => 10,
  'mail_smtpauthtype' => 'LOGIN',
  'trusted_domains' =>
  array (
    0 => 'XXX',
    1 => 'XXX',
  ),
  'mail_from_address' => 'donotreply',
  'mail_domain' => 'XXX',
  'secret' => 'XXX',
  'forceSSLforSubdomains' => true,
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'trashbin_retention_obligation' => 'auto',
  'versions_retention_obligation' => '90, auto',
  'appcodechecker' => true,
  'enabledPreviewProviders' => array(
        'OC\Preview\PNG',
        'OC\Preview\JPEG',
        'OC\Preview\GIF',
        'OC\Preview\BMP',
        'OC\Preview\XBitmap',
        'OC\Preview\MP3',
        'OC\Preview\MarkDown'
  ),
  'cipher' => 'AES-256-CFB',
  'filelocking.enabled' => true,
);

**Are you using external storage, if yes which one:** local/smb/sftp/
no

**Are you using encryption:**
yes

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/
no

### Client configuration
**Browser:**
Mozilla Firefox 41.0.2 (x86)

**Operating system:**
Windows 7 64bit
#### Web server error log
---

#### ownCloud log (data/owncloud.log)
Debug	files_locking	PASS: Acquired read lock for /var/www/owncloud/data/USER/files_encryption/keys/files/Public/test.docx/OC_DEFAULT_MODULE/fileKey	2015-11-02T18:23:52+01:00
Debug	files_locking	INFO: Read lock requested for /var/www/owncloud/data/USER/files_encryption/keys/files/Public/test.docx/OC_DEFAULT_MODULE/fileKey	2015-11-02T18:23:52+01:00
Debug	files_locking	INFO: Incrementing lock type 1 count for /var/www/owncloud/data/USER/files/Public/test.docx	2015-11-02T18:23:52+01:00

#### Browser log
---

### PR DESCRIPTION
**Pull request by schiessle**
Make sure that we update the unencrypted size for versions.

fix https://github.com/owncloud/core/issues/20234

Steps to test:
- Enable encryption
- Upload a larger file (pdf, odt, jpg,...)
- Upload another copy of the same file
- check if you can download/revert the versions

@icewind1991 please also have a look at it. I fixed this now in the encryption storage wrapper. But I have the feeling that filesystem/cache should take care about copying the file cache data if the file was just copied instead of re-scanning the file.

### CODE REVIEW

**Comment by schiesbn** [comment]
@karlitschek should be backported to 8.2, 8.1 is not affected.

File: tests/lib/files/storage/wrapper/encryption.php [hunk 1]
```diff
@@ -560,6 +560,102 @@ public function testCopyBetweenStorage($encryptionEnabled, $mountPointEncryption
...
+			[
+				$sourceStorage,
+				$sourceInternalPath,
+				$targetInternalPath,
+				false,
+				false
+			]
+		);
+
+		$this->assertSame($copyResult, $result);
```
Comment (DeepDiver1975) [inline tests/lib/files/storage/wrapper/encryption.php]: @schiesbn we should assert the actual size as well - right?

Comment (schiesbn) [inline tests/lib/files/storage/wrapper/encryption.php]: The method only returns if the copy operation was successful or not.
In line 620 I make sure that we call "updateUnencryptedSize" with the correct size.

Comment (DeepDiver1975) [inline tests/lib/files/storage/wrapper/encryption.php]: great - I did not see that :+1:

**Comment by schiesbn** [comment]
failing Oracle test unrelated, please ignore

**Comment by karlitschek** [comment]
backport makes sense :+1:

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Pull request merged**